### ISSUE TITLE
Concat Vendor CSS with less and copy associated images to assets...

### ISSUE THREAD

**Issue by cymantic** [ngbp/ngbp#95]
When trying to include vendor CSS, that also refers to images provided by the vendor (from jquery fancybox)....

I want the images and css to be in the same folder during a build, so that the css when compiled also references the same image paths.

I can add a task to the grunt build file to copy the vendor images to the appropriate location in assets (assets/vendor-image.png)

The CSS files needs to be in the same location (assets/vendor.css), or needs to be joined to the output of less.

**With less v1.4.1 I could @import the css file, however recess requires less 1.3**.

Once recess is updated, then I can import css like other vendor less files, until then I'm manually putting the files (CSS and images) into assets and adding a link in the index.html

Any other options?

**Comment by joshdmiller** [comment]
I think the best way to deal with this case is through #57. That will allow you to specify a CSS file to include statically, which would mean just throwing it into `assets/`. In this way, it would also be concatenated with your other CSS and included automatically.

Does that sound about right?

**Comment by artworkad** [comment]
@joshdmiller when I include the css in main.less like

```
@import '../assets/font-awesome/less/font-awesome.less';
```

how will the fonts be included? Fonts are in /assets/font-awesome/font

**Comment by cymantic** [comment]
font-awesome allows you to set a less variable...
put the font files into your assets folder (assets/fonts/...)
and then add to your variables.less file:

`@FontAwesomePath: "font";`

if you don't wan't to copy the font files to assets you can add this to the Gruntfile.js as part of the copy:build_assets

```
{
            src: ['**'],
            dest: '<%= build_dir %>/assets/font',
            cwd: 'vendor/font-awesome/font',
            expand: true
}
```

**Comment by artworkad** [comment]
@cymantic looking good, but font-awesome is looking for the fonts in /build/font, instead of /build/assets/font any ideas?

**Comment by cymantic** [comment]
in main.less I'm including font-awesome from the vendor folder not the assets...
`@import '../../vendor/font-awesome/less/font-awesome.less';`

**Comment by artworkad** [comment]
@cymantic yeah I do it the same way, but when I look at chrome console it font-awesome tries to get the fonts in /build/font but they are in /build/assets/font. When I change to

```
     {  
        src: ['**'],
        dest: '<%= build_dir %>/font',
        cwd: 'vendor/font-awesome/font',
        expand: true
      }
```

it works but the fonts are not in assets anymore. Any idea?

**Comment by cymantic** [comment]
@ArtworkAD check that your variable fontawesomepath is just **font**, it's relative to the compiled css file, if font is in assets and the css is then there is no **/** in front of **font**

**Comment by joshdmiller** [comment]
I'm a little confused. Why are you including the font files statically? Wouldn't it be better to load them from CDN?

**Comment by cymantic** [comment]
Yes it would be better to load these from a CDN as set up in ng-boilerplate by default, but not everything is able to run with internet access. I wouldn't suggest changing to this unless you have a good reason. (and this is a side issue to the main one of getting the vendor css concatenated with appropriate relative URLs for CSS referenced assets)

**Comment by strangfeld** [comment]
I have inserted this just between the copy:build_vendorjs and copy:compile_assets task.

```
build_vendorcss: {
    files: [
        {
            src: [ '<%= vendor_files.css %>' ],
            dest: '<%= build_dir %>/',
            cwd: '.',
            expand: true
        }
    ]
},
```

And of course adding it beneath in the complete build task, which looks like this :

```
grunt.registerTask( 'build', [
  'clean', 'html2js', 'jshint','recess:build',
  'copy:build_assets', 'copy:build_appjs', 'copy:build_vendorjs', 'copy:build_vendorcss',
  'index:build', 'karmaconfig', 'karma:continuous' 
]);
```

Excuse my formatting, this is my first comment on anything on github EVER ! I don't know if it's useful, but the css appears now in the build/vendor directory. Hope it helps anyone.

**Comment by joshdmiller** [comment]
Closed by 29502bff8eb649501945aef37fe50d81c36151ba.

**Comment by thegrymek** [comment]
Hi, 

This one worked for me

https://github.com/ngbp/ngbp/pull/336/files?diff=unified
